torchmultimodal/models/clip/model.py

```diff
@@ -88,85 +108,121 @@ def clip_vit_l14() -> CLIP:
+    clip = CLIP(vision_encoder, text_encoder)
+    if pretrained:
+        load_module_from_url(clip, CLIP_MODEL_MAPPING["rn50"])
+    return clip
 
 
-def clip_rn101() -> CLIP:
+def clip_rn101(pretrained: bool = False) -> CLIP:
     vision_encoder = ResNetForCLIP(
         layers=(3, 4, 23, 3),
-        output_dim=1024,
-        heads=1024,
-        width=2048,
+        output_dim=512,
+        heads=32,
+        width=64,
     )
-    text_encoder = CLIPTextEncoder(embedding_dim=1024)
-    return CLIP(vision_encoder, text_encoder)
+    text_encoder = CLIPTextEncoder(embedding_dim=512)
+    clip = CLIP(vision_encoder, text_encoder)
+    if pretrained:
+        load_module_from_url(clip, CLIP_MODEL_MAPPING["rn101"])
+    return clip
 
 
 # Note: these models require larger image sizes
-def clip_rn50x4() -> CLIP:
+def clip_rn50x4(pretrained: bool = False) -> CLIP:
     vision_encoder = ResNetForCLIP(
         layers=(4, 6, 10, 6),
         output_dim=640,
-        heads=1280,
+        heads=40,
         input_resolution=288,
-        width=2560,
+        width=80,
+    )
+    text_encoder = CLIPTextEncoder(
+        embedding_dim=640, width=640, dim_feedforward=2560, heads=10
     )
-    text_encoder = CLIPTextEncoder(embedding_dim=1024, width=640, heads=12)
-    return CLIP(vision_encoder, text_encoder)
+    clip = CLIP(vision_encoder, text_encoder)
+    if pretrained:
+        load_module_from_url(clip, CLIP_MODEL_MAPPING["rn50x4"])
+    return clip
 
 
-def clip_rn50x16() -> CLIP:
+def clip_rn50x16(pretrained: bool = False) -> CLIP:
     vision_encoder = ResNetForCLIP(
         layers=(6, 8, 18, 8),
         output_dim=768,
-        heads=1536,
+        heads=48,
         input_resolution=384,
-        width=3072,
+        width=96,
     )
-    text_encoder = CLIPTextEncoder(embedding_dim=768, width=768, heads=12)
-    return CLIP(vision_encoder, text_encoder)
+    text_encoder = CLIPTextEncoder(
+        embedding_dim=768, width=768, dim_feedforward=3072, heads=12
+    )
+    clip = CLIP(vision_encoder, text_encoder)
+    if pretrained:
+        load_module_from_url(clip, CLIP_MODEL_MAPPING["rn50x16"])
+    return clip
 
 
-def clip_rn50x64() -> CLIP:
+def clip_rn50x64(pretrained: bool = False) -> CLIP:
     vision_encoder = ResNetForCLIP(
         layers=(3, 15, 36, 10),
         output_dim=1024,
-        heads=2048,
+        heads=64,
         input_resolution=448,
-        width=4096,
+        width=128,
+    )
+    text_encoder = CLIPTextEncoder(
+        embedding_dim=1024, width=1024, dim_feedforward=4096, heads=16
     )
-    text_encoder = CLIPTextEncoder(embedding_dim=1024, width=1024, heads=16)
-    return CLIP(vision_encoder, text_encoder)
+    clip = CLIP(vision_encoder, text_encoder)
+    if pretrained:
+        load_module_from_url(clip, CLIP_MODEL_MAPPING["rn50x64"])
+    return clip
 
 
 # Note: these models use torchvision's ResNet
-def clip_rn50_tv() -> CLIP:
+def clip_rn50_tv(pretrained: bool = False) -> CLIP:
```

Comment:
remove the pretrained parameter to all tv ones, its confusing

Reply:
I was trying to keep the "api" consistent for all the model functions. If someone had some kind of model factory code or config I could also see it being annoying that only some models have the pretrained flag.

Reply:
in this case, I would say, functionality is more important than consistency. why would we expose an arg that does nothing (and gives an impression its doing something if the user misses the warning log)